swagger_spec_validator/schemas/v2.0/schema.json

```diff
@@ -1360,6 +1360,9 @@
                 "minimum": {
                     "$ref": "custom_json_schema.json#/properties/minimum"
                 },
+                "listSubtype": {
+                    "$ref": "custom_json_schema.json#/properties/listSubtype"
+                },
```

Comment:
Just making sure we don't want to override this in our custom schema somehow instead of modifying the base. If that's even possible.

Reply:
I kind have two answers here.  These are different sections. Our custom_json_schema.json is our JSON spec not our open_api 2.0 spec.     The schema.json is our open_api 2.0 spec which depends on some defs from JSON spec.  They are different, since other things could use the JSON spec. 

The section that is defining listSubtype is the schema def for open api2.0.  If we wanted we could move this entire section to a new file. called custom_schema_def.json or something like that and point schema.json to it for the def. But we probably don't want put this in our custom_json_schema.json since they are different things. Plus once we move to 3.0 we probably can just point 3.0 schema to our custom_json_schema.json and have no issues. This listSubtype added constraint might give us a little trouble having to remember to add it back. Thoughts on where you think this should live?

Reply:
I'm not sure I understand the difference between the "JSON spec" and the "open_api 2.0 spec" (which is also JSON, I think). Could you elaborate on that?

Reply:
I might be understanding it wrong too but here is how i view it.

this explains kind of how others use the http://json-schema.org/draft-04/schema  as well.

https://www.js-data.io/docs/validation

So http://json-schema.org/draft-04/schema is the base schema that JSON data must follow.  Other specs can use this as well besides swagger open-api 2.0. Open api 2.0 uses the spec as its base and adds on top of it.

  So when 3.0 comes out there is a good chance that the base JSON spec http://json-schema.org/draft-04/schema will not change. open-api 3.0 will add its own new stuff.    What I did is just took that external file and made it internal and added a few things. Keeping the separation between the two like spec_validator did. So when 3.0 or changes happen in 2.0. All we have to do is point the base back to our custom version of  http://json-schema.org/draft-04/schema. We really could have combined these two files as well. Or made more files definining other defs as well.

Reply:
> So http://json-schema.org/draft-04/schema is the base schema that JSON data must follow.

http://json-schema.org/draft-04/schema is the base schema that *JSON schemas* must follow. So it's a schema for schemas (very meta).

And then from there, `swagger_spec_validator/schemas/v2.0/schema.json` is the v2.0 schema of Open API, specified using the JSON Schema format.

So it looks like we've essentially forked both the JSON Schema standard and the Open API standard. So much for standards! Your changes make sense now. I guess I'm wondering if we could avoid modifying the base JSON Schema spec, though, by changing our version of the Open API spec to say "use the types found in the JSON Schema spec + these custom ones defined in this file" instead. Does that make sense?